clients/java/src/test/java/io/zeebe/client/job/ThrowErrorTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.job;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.client.util.ClientTest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ThrowErrorRequest;
+import java.time.Duration;
+import org.junit.Test;
+
+public class ThrowErrorTest extends ClientTest {
+
+  @Test
+  public void shouldThrowError() {
+    // given
+    final long jobKey = 12;
+    final String errorCode = "errorCode";
+    final String errorMsg = "errorMsg";
+
+    // when
+    client.newThrowErrorCommand(jobKey).errorCode(errorCode).errorMessage(errorMsg).send().join();
+
+    // then
+    final ThrowErrorRequest request = gatewayService.getLastRequest();
+    assertThat(request.getJobKey()).isEqualTo(jobKey);
+    assertThat(request.getErrorCode()).isEqualTo(errorCode);
+    assertThat(request.getErrorMessage()).isEqualTo(errorMsg);
```

Comment:
Since the error message is an optional parameter, we should have a test case with and without it.